### PR TITLE
Credentials update

### PR DESCRIPTION
**Pull request by susesgartner**
Update cloud credentials functions for every provider.

### CODE REVIEW

File: extensions/machinepools/digitalocean_machine_config.go
Comment (slickwarren) [inline extensions/machinepools/digitalocean_machine_config.go:52]: can you double check this, I'm pretty sure it is supposed to be blank for the machine / node templates. Otherwise, you'd need to update the other files in this folder that set it to blank for say linode.

File: extensions/cloudcredentials/aws/create.go
Comment (slickwarren) [inline extensions/cloudcredentials/aws/create.go:31]: if there's a way to make this global for all the cloudcredentials that would be nice.

**Review by slickwarren — COMMENTED** [review]
also ping'd offline to resolve something

File: extensions/cloudcredentials/aws/create.go
Comment (susesgartner) [reply extensions/cloudcredentials/aws/create.go:31]: Updated!

File: extensions/cloudcredentials/aws/create.go
Comment (slickwarren) [inline extensions/cloudcredentials/aws/create.go:36]: is the 0th user always the one you expect? In the past, we've run into trouble with how norman sorts things

File: extensions/cloudcredentials/digitalocean/create.go
Comment (slickwarren) [inline extensions/cloudcredentials/digitalocean/create.go:40]: confirming this name is correct; since DO uses a token, not json afaik

File: extensions/rke1/nodetemplates/aws/create.go
Comment (slickwarren) [inline extensions/rke1/nodetemplates/aws/create.go:21]: why is the fileKey loaded 2x?

**Review by slickwarren — COMMENTED** [review]
a few nits; waiting for jenkins runs for approval

File: extensions/cloudcredentials/azure/create.go
Comment (markusewalker) [inline extensions/cloudcredentials/azure/create.go:36]: Same comment as the AWS above.

File: extensions/cloudcredentials/cloudcredentials.go
Comment (markusewalker) [inline extensions/cloudcredentials/cloudcredentials.go:35]: nit - can you add a godoc here?

File: extensions/cloudcredentials/cloudcredentials.go
Comment (markusewalker) [inline extensions/cloudcredentials/cloudcredentials.go:43]: nit - can you add a new line  after 42? So that the return is not so consolidated? Same for the rest in this switch statement.

File: extensions/cloudcredentials/cloudcredentials.go
Comment (markusewalker) [inline extensions/cloudcredentials/cloudcredentials.go:82]: Is there a specific reason we are `panic`? Couldn't we just do a `logrus.Errorf` or something and not panic?

**Review by markusewalker — COMMENTED** [review]
Left some comments, will wait for new jobs to kick off.

File: extensions/cloudcredentials/aws/create.go
Comment (igomez06) [inline extensions/cloudcredentials/aws/create.go:23]: I agree with Markus and Caleb, getting the full list of users and then assuming it's the first entry sounds like would break when multiple users have been created. Perhaps we pass in the user data as well?

File: extensions/cloudcredentials/cloudcredentials.go
Comment (igomez06) [reply extensions/cloudcredentials/cloudcredentials.go:82]: I think Sam is good here, if the user provides a non-supported value we should panic.

**Review by igomez06 — COMMENTED** [review]
overall looks good, I have the same concerns as Markus and Caleb. Would definitely need see a run with all providers to provide a sign off.

File: extensions/cloudcredentials/aws/create.go
Comment (susesgartner) [reply extensions/cloudcredentials/aws/create.go:36]: It was in the case of admin/standard clients. That being said I do wonder if there is a better way to pull this ID without

File: extensions/rke1/nodetemplates/aws/create.go
Comment (susesgartner) [reply extensions/rke1/nodetemplates/aws/create.go:21]: The first load is for the nodeTemplateConfig, the second is for the Credentials.

File: extensions/cloudcredentials/aws/create.go
Comment (igomez06) [reply extensions/cloudcredentials/aws/create.go:36]: I think we should pass the user ID instead. How do we guarantee that the first element of the user's list is the user that created the credential?

**Review by igomez06 — CHANGES_REQUESTED** [review]
Better way of getting the Creator ID

File: extensions/cloudcredentials/aws/create.go
Comment (susesgartner) [reply extensions/cloudcredentials/aws/create.go]: This has been fixed!

**Review by markusewalker — DISMISSED** [review]

**Review by igomez06 — APPROVED** [review]
Looks great.

**Comment by igomez06** [comment]
can we get back ports too?

**Comment by susesgartner** [comment]
I'll add the backports once code freeze is lifted.

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Comment by markusewalker** [comment]
Thanks @susesgartner ! If possible, please hold off on merging until the freeze is lifted and the backports can be merged alongside this.

**Review by slickwarren — APPROVED** [review]

**Comment by caliskanugur** [comment]
Backports:

- https://github.com/rancher/shepherd/pull/280
- https://github.com/rancher/shepherd/pull/281
- https://github.com/rancher/shepherd/pull/282

**Review by markusewalker — APPROVED** [review]

**Pull request merged**
